python/ray/tune/schedulers/hb_bohb.py

```diff
@@ -97,10 +97,20 @@ def on_trial_result(
         if not bracket.filled() or any(
             status != Trial.PAUSED for t, status in statuses if t is not trial
         ):
+            # BOHB Specific. This hack existed in old Ray versions
+            # and was removed, but it needs to be brought back
+            # as otherwise the BOHB doesn't behave as intended.
```

Comment:
```suggestion
            # as otherwise the BOHB doesn't behave as intended.
            # The default concurrency limiter works by discarding
            # new suggestions if there are more running trials
            # than the limit. That doesn't take into account paused
            # trials. With BOHB, this leads to N trials finishing
            # completely and then another N trials starting,
            # instead of trials being paused and resumed in brackets
            # as intended.
```